### ISSUE TITLE
what is the difference between DDP and single-GPU for profiler?

### ISSUE THREAD

**Issue by huangfu170** [Lightning-AI/pytorch-lightning#20135]
### 📚 Documentation

I use DDP strategy with Pytorch profiler to record the trace. I use the following code, is there something different between DDP and single-GPU for profiler?
`
profiler=PyTorchProfiler(
    on_trace_ready=torch.profiler.tensorboard_trace_handler("lightning_logs/profiler/version_0"),
    schedule=torch.profiler.schedule(wait=1, warmup=1, active=5)
)
tb_logger=pl.loggers.TensorBoardLogger('./lightning_logs',version=0,name='profiler')
trainer=pl.Trainer(max_epochs=100,accelerator='gpu',devices=[0,1,2],
                   profiler=profiler,
                   strategy='ddp',
                   callbacks=[early_stop_callback,model_checkpoint_callback],
                   gradient_clip_val=1,log_every_n_steps=2,
                   logger=tb_logger)
trainer.fit()
`
And I found 6 files in my log folder, for every workers I found two spans, In the span 1, there's no traces at all. In the span 2, the trace is found, but only trace for validation_step in it. By the way, there is no problem in single-GPU training, the optimizer_step will be recorded.
![图片](https://github.com/user-attachments/assets/d980329c-9a61-4418-a37a-51875f3fdc3f)
 
![图片](https://github.com/user-attachments/assets/0974433f-66f5-4e10-a780-81e470e531f0)
![图片](https://github.com/user-attachments/assets/33418ab2-d95c-46db-87d3-2cb5ee1ff442)

cc @borda

**Comment by huangfu170** [comment]
@awaelchli
